crypto/crypto/src/x25519.rs

```diff
@@ -1,340 +1,151 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-//! An implementation of x25519 elliptic curve key pairs required for
-//! [Diffie-Hellman key
-//! exchange](https://en.wikipedia.org/wiki/Diffie%E2%80%93Hellman_key_exchange)
+//! An abstraction of x25519 elliptic curve keys required for
+//! [Diffie-Hellman key exchange](https://en.wikipedia.org/wiki/Diffie%E2%80%93Hellman_key_exchange)
 //! in the Libra project.
-//!
-//! This is an API for [Elliptic Curves for Security - RFC
-//! 7748](https://tools.ietf.org/html/rfc7748) and which deals with
-//! long-term key generation and handling (`X25519StaticPrivateKey`,
-//! `X25519StaticPublicKey`) as well as short-term keys (`X25519EphemeralPrivateKey`,
-//! `X25519PublicKey`).
-//!
-//! The default type for a Diffie-Hellman secret is an ephemeral
-//! one, forming a `PrivateKey`-`PublicKey` pair with `X25519Publickey`,
-//! and is not serializable, since the use of fresh DH secrets is
-//! recommended for various reasons including PFS.
-//!
-//! We also provide a "static" implementation `X25519StaticPrivateKey`,
-//! which supports serialization, forming a `PrivateKey`-`PublicKey` pair
-//! with `X25519StaticPublickey`. This later type is precisely a
-//! [newtype](https://doc.rust-lang.org/1.5.0/style/features/types/newtype.html)
-//! wrapper around `X25519PublicKey`, to which it coerces through `Deref`.
+//! Ideally, only `x25519::PrivateKeyBytes` and `x25519::PublicKeyBytes` should be used throughout the
+//! codebase, until the bytes are actually used in cryptographic operations.
 //!
 //! # Examples
 //!
 //! ```
-//! use libra_crypto::x25519::*;
+//! use libra_crypto::x25519;
 //! use rand::{rngs::StdRng, SeedableRng};
 //!
-//! // Derive an X25519 static key pair from seed using the extract-then-expand HKDF method from RFC 5869.
-//! let salt = &b"some salt"[..];
-//! // In production, ensure seed has at least 256 bits of entropy.
-//! let seed = [5u8; 32]; // seed is denoted as IKM in HKDF RFC 5869.
-//! let info = &b"some app info"[..];
-//!
-//! let (private_key1, public_key1) = X25519StaticPrivateKey::derive_keypair_from_seed(Some(salt), &seed, Some(info));
-//! let (private_key2, public_key2) = X25519StaticPrivateKey::derive_keypair_from_seed(Some(salt), &seed, Some(info));
-//! assert_eq!(public_key1, public_key2);
-//!
-//! // Generate a random X25519 ephemeral key pair from an RNG (in this example a StdRng)
-//! use libra_crypto::Uniform;
+//! // Derive an X25519 private key for testing.
 //! let seed = [1u8; 32];
 //! let mut rng: StdRng = SeedableRng::from_seed(seed);
-//! let private_key = X25519StaticPrivateKey::generate(&mut rng);
-//! let public_key: X25519StaticPublicKey = (&private_key).into();
+//! let private_key = x25519::PrivateKeyBytes::for_test(&mut rng);
```

Comment:
recommend you call it X25519PrivateKey to prevent name collisions alternatively we import x25519 and use x22519::PrivateKey everywhere I prefer the former.

Reply:
I prefer the latter which is what I did : o

Reply:
This makes our crypto crate rather ... awkward looking. Please, let's be consistent.

Reply:
I'd argue that namespaces are here exactly for this: to disambiguate and prevent name collisions. I'd be happy to unify the rest of the crate :P

Reply:
I've got a 3rd option for you, while we're on the topic of bikesheding:
`PrivateKey<X25519>`

Reply:
pls no : D